sql/src/main/java/io/crate/operation/collect/ShardCollectorProviderFactory.java

```diff
@@ -0,0 +1,87 @@
+ * agreement.
+ */
+
+package io.crate.operation.collect;
+
+import io.crate.blob.v2.BlobIndicesService;
+import io.crate.blob.v2.BlobShard;
+import io.crate.executor.transport.TransportActionProvider;
+import io.crate.lucene.LuceneQueryBuilder;
+import io.crate.metadata.Functions;
+import io.crate.metadata.Schemas;
+import io.crate.operation.NodeJobsCounter;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.index.shard.IndexShard;
+import org.elasticsearch.threadpool.ThreadPool;
+
+import static io.crate.blob.v2.BlobIndex.isBlobIndex;
+
+public class ShardCollectorProviderFactory {
+
+    private final Schemas schemas;
+    private final ClusterService clusterService;
+    private final ThreadPool threadPool;
+    private final TransportActionProvider transportActionProvider;
+    private final BlobIndicesService blobIndicesService;
+
+    private final Functions functions;
+    private final LuceneQueryBuilder luceneQueryBuilder;
+    private final NodeJobsCounter nodeJobsCounter;
+    private final BigArrays bigArrays;
+    private final Settings settings;
+
+    public ShardCollectorProviderFactory(ClusterService clusterService,
+                                         Settings settings,
+                                         Schemas schemas,
+                                         ThreadPool threadPool,
+                                         TransportActionProvider transportActionProvider,
+                                         BlobIndicesService blobIndicesService,
+                                         Functions functions,
+                                         LuceneQueryBuilder luceneQueryBuilder,
+                                         NodeJobsCounter nodeJobsCounter,
+                                         BigArrays bigArrays) {
+        this.settings = settings;
+        this.schemas = schemas;
+        this.clusterService = clusterService;
+        this.threadPool = threadPool;
+        this.transportActionProvider = transportActionProvider;
+        this.blobIndicesService = blobIndicesService;
+        this.functions = functions;
+        this.luceneQueryBuilder = luceneQueryBuilder;
+        this.nodeJobsCounter = nodeJobsCounter;
+        this.bigArrays = bigArrays;
+    }
+
+    public ShardCollectorProvider create(IndexShard indexShard) {
+        if (isBlobIndex(indexShard.shardId().getIndexName())) {
+            BlobShard blobShard = blobIndicesService.blobShardSafe(indexShard.shardId());
+            return new BlobShardCollectorProvider(blobShard, clusterService, nodeJobsCounter, functions,
+                threadPool, settings, transportActionProvider, bigArrays);
+        } else {
+            return new LuceneShardCollectorProvider(
+                schemas, luceneQueryBuilder, clusterService, nodeJobsCounter, functions,
```

Comment:
Usually we format either all args in one line or one arg per line. Would be nice to keep this consistent.

Reply:
jfyi: afaik this style is widely used all over our codebase and I wasn't aware that this violates our style-guide. I'm fine with changing it, but we should fix/agree on it.